pulp_ostree/tests/functional/conftest.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+import uuid
+
+from pulp_ostree.tests.functional.utils import gen_ostree_remote
+from pulpcore.client.pulp_ostree import (
+    ApiClient,
+    ContentCommitsApi,
+    ContentRefsApi,
+    RepositoriesOstreeApi,
+    RepositoriesOstreeVersionsApi,
+    RepositorySyncURL,
+    RemotesOstreeApi,
+)
+
+
+@pytest.fixture()
+def ostree_client(_api_client_set, bindings_cfg):
+    """Fixture to provide a client to Pulp API"""
+    api_client = ApiClient(bindings_cfg)
+    _api_client_set.add(api_client)
+    yield api_client
+    _api_client_set.remove(api_client)
+
+
+@pytest.fixture()
+def ostree_content_refs_api(ostree_client):
+    """Fixture that returns an instance of ContentRefsApi"""
+    return ContentRefsApi(ostree_client)
+
+
+@pytest.fixture()
+def ostree_repositories_api(ostree_client):
+    """Fixture that returns an instance of RepositoriesOstreeApi"""
+    return RepositoriesOstreeApi(ostree_client)
+
+
+@pytest.fixture()
+def ostree_remotes_api(ostree_client):
+    """Fixture that returns an instance of RepositoriesOstreeApi"""
+    return RemotesOstreeApi(ostree_client)
+
+
+@pytest.fixture()
+def ostree_repositories_versions_api(ostree_client):
+    """Fixture that returns an instance of RepositoriesOstreeVersionsApi"""
+    return RepositoriesOstreeVersionsApi(ostree_client)
+
+
+@pytest.fixture()
+def ostree_repository_sync_url(ostree_client):
+    """Fixture that returns an instance of RepositorySyncURL"""
+    return RepositorySyncURL(ostree_client)
+
+
+@pytest.fixture()
+def ostree_content_commits_api(ostree_client):
+    """Fixture that returns an instance of ContentCommitsApi"""
+    return ContentCommitsApi(ostree_client)
+
+
+@pytest.fixture()
+def remote(ostree_remotes_api):
+    """Fixture that creates an ostree Remote and returns the object created"""
+    body = gen_ostree_remote(depth=0, policy="immediate")
+    return ostree_remotes_api.create(body)
```

Comment:
Instead of writing this, you may want to implement a factory that will handle the setup and cleanup of resources by itself by leveraging `gen_object_with_cleanup`, like so: https://github.com/pulp/pulp_gem/blob/4de8c2adf1be973958d95b16f0c4137dab7ba066/pulp_gem/tests/functional/conftest.py#L105-L116. At the moment, we do not support domains, so you can skip that part.

Do you think it makes sense to follow the same concept for the corresponding `repo` and `distribution` fixture?